remote_attestation/java/src/com/google/oak/remote_attestation/Message.java

```diff
@@ -0,0 +1,356 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package com.google.oak.remote_attestation;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInput;
+import java.io.DataInputStream;
+import java.io.DataOutput;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import static java.lang.Math.toIntExact;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+/** Wrapper class for Remote Attestation Protocol messages. */
+public class Message {
+  static final int CLIENT_HELLO_HEADER = 1;
```

Comment:
By header do you mean frame type? If so, why not use the same terminology as SC?

Reply:
Header is the terminology used in the SC document.

Reply:
You are right, even in the implementation they use both frame header and frame type. Maybe we can all pick one at some point.